lib/filterx/filterx-error.c

```diff
@@ -98,3 +98,15 @@ filterx_error_set_info(FilterXError *error, gchar *info, gboolean free_info)
   error->info = info;
   error->free_info = free_info;
 }
+
+void
+filterx_error_copy(FilterXError *src, FilterXError *dst)
+{
+  g_assert(!dst->message);
+
+  dst->message = src->message;
+  dst->expr = src->expr;
+  dst->object = filterx_object_ref(src->object);
+  dst->info = src->free_info ? g_strdup(src->info) : src->info;
```

Comment:
If this is not on a fastpath, I think we should always strdup() the info. Sometimes it's a borrowed reference, but it may not survive the copy.

Reply:
Done.